extensions/basic_auth/plugin.cc

```diff
@@ -0,0 +1,192 @@
+std::unordered_map<std::string, S> configurationData;
+
+bool PluginRootContext::onConfigure(size_t size) {
+  // Parse configuration JSON string.
+  if (size > 0 && !configure(size)) {
+    LOG_WARN("configuration has errors, but initialzation can continue.");
+  }
+  return true;
+}
+
+bool PluginRootContext::configure(size_t configuration_size) {
+  auto configuration_data = getBufferBytes(WasmBufferType::PluginConfiguration,
+                                           0, configuration_size);
+  // Parse configuration JSON string.
+  auto result = ::Wasm::Common::JsonParse(configuration_data->view());
+  if (!result.has_value()) {
+    LOG_WARN(absl::StrCat("cannot parse plugin configuration JSON string: ",
+                          configuration_data->view()));
+    return false;
+  }
+  // j is a JsonObject holds configuration data
+  auto j = result.value();
+
+  if (!JsonArrayIterate(j, "pathPrefixes", [&](const json& prefixes) -> bool {
+        auto pathPrefix =
+            JsonGetField<std::string>(prefixes, "pathPrefix").value_or("");
+        if (pathPrefix == "") return false;
+        struct S s1;
+        if (!JsonArrayIterate(prefixes, "requestMethods",
+                              [&](const json& methods) -> bool {
+                                s1.requestMethods.insert(std::string(methods));
+                                return true;
+                              })) {
+          LOG_WARN("Failed to parse configuration for request methods.");
+          return false;
+        }
+
+        if (!JsonArrayIterate(prefixes, "credentials",
+                              [&](const json& credentials) -> bool {
+                                auto cred = std::string(credentials);
+                                s1.encodedCredentials.insert(
+                                    Base64::encode(cred.data(), cred.size()));
+                                return true;
+                              })) {
+          LOG_WARN("Failed to parse configuration for credentials.");
+          return false;
+        }
+        auto checkPath = configurationData.find(pathPrefix);
+        if (checkPath != configurationData.end()) {
+          for (auto& methods : s1.requestMethods) {
+            configurationData[pathPrefix].requestMethods.insert(methods);
+          }
+
+          for (auto& creds : s1.encodedCredentials) {
+            configurationData[pathPrefix].encodedCredentials.insert(creds);
+          }
+        } else {
+          configurationData[pathPrefix] = s1;
+        }
+        return true;
+      })) {
+    LOG_WARN("Failed to parse configuration information.");
+  }
+
+  return true;
+}
+
+FilterHeadersStatus PluginContext::onRequestHeaders(uint32_t, bool) {
+  auto authorizationHeader = getRequestHeader("authorization")->toString();
+  auto pathPrefix = getRequestHeader(":path")->toString();
+  auto method = getRequestHeader(":method")->toString();
+  auto checkPathPrefix = configurationData.find(pathPrefix);
```

Comment:
This checks the whole path against prefix in configuration... Please make sure the logic here is right.. this is going to be a very sensitive code path since it relates to security.

Reply:
Yes, you need to iterate over match conditions and compare for a prefix. This would be good enough for now. TRIE would be faster.

Reply:
Oh i see, i just realize that the naming is misleading. In fact, this logic covers the case where the `match_pattern` is exact.